app/controllers/tasks_controller.rb

```diff
@@ -4,31 +4,38 @@ class TasksController < ApplicationController
   # GET /tasks
   # GET /tasks.json
   def index
-    @tasks = Task.all
+    @list = List.find(params[:list_id])
+    @tasks = @list.tasks.new
   end
 
   # GET /tasks/1
   # GET /tasks/1.json
   def show
+    @list = List.find(params[:list_id])
+    @tasks = @list.tasks.all
   end
 
   # GET /tasks/new
   def new
-    @task = Task.new
+    @list = List.find(params[:list_id])
+    @task = @list.tasks.new
   end
 
   # GET /tasks/1/edit
   def edit
+    @list = List.find(params[:list_id])
+    @task = @list.tasks.find(params[:task_id])
```

Comment:
I'm running into an error when trying to edit my tasks, and thoughts on why this is happening?

<img width="1021" alt="screenshot 2018-09-26 11 25 07" src="https://user-images.githubusercontent.com/3793148/46090585-085b2b00-c17f-11e8-8b13-dab2b2f18576.png">